tests/unittest_scoped_nodes.py

```diff
@@ -473,9 +473,19 @@ def test_argnames(self) -> None:
         astroid = builder.parse(code, __name__)
         self.assertEqual(astroid["f"].argnames(), ["a", "b", "c", "args", "kwargs"])
 
-        code_with_kwonly_args = "def f(a, b, *, c=None, d=None): pass"
+        code_with_kwonly_args = "def f(a, b, *args, c=None, d=None, **kwargs): pass"
         astroid = builder.parse(code_with_kwonly_args, __name__)
-        self.assertEqual(astroid["f"].argnames(), ["a", "b", "c", "d"])
+        self.assertEqual(
+            astroid["f"].argnames(), ["a", "b", "args", "c", "d", "kwargs"]
+        )
+
+    @unittest.skipUnless(PY38_PLUS, "positional-only argument syntax")
+    def test_positional_only_argnames(self) -> None:
+        code = "def f(a, b, /, c=None, *args, d, **kwargs): pass"
+        astroid = builder.parse(code, __name__)
+        self.assertEqual(
+            astroid["f"].argnames(), ["a", "b", "c", "args", "d", "kwargs"]
+        )
```

Comment:
Interesting that positional-only arguments are included 🤔
Wasn't sure about that, since the method uses `args.arguments` when there is `args.posonlyargs` as well. It does make sense however to include all names here. A check if an arg is positional-only should be done separately (in pylint).

Maybe it would be good to update the docstring a bit?
```
[...] as well as positional-only and keyword-only arguments.
```